daemon/analytics.c

```diff
@@ -847,6 +847,28 @@ void set_global_environment()
     setenv("HOME", verify_required_directory(netdata_configured_home_dir), 1);
     setenv("NETDATA_HOST_PREFIX", netdata_configured_host_prefix, 1);
 
+    {
+        size_t len = 0;
+
+        for (size_t i = 1; i < PLUGINSD_MAX_DIRECTORIES && plugin_directories[i]; i++) {
+            if (i > 1)
+                len++;
+            len += strlen(plugin_directories[i]);
+        }
+
+        char *user_plugins_dir = callocz(1, len + 1);
+
+        for (size_t i = 1; i < PLUGINSD_MAX_DIRECTORIES && plugin_directories[i]; i++) {
+            if (i > 1)
+                strcat(user_plugins_dir, " ");
+            strcat(user_plugins_dir, plugin_directories[i]);
```

Comment:
Normal implementations of `strcat` (please, see [this](https://github.com/lattera/glibc/blob/master/string/strcat.c) and [this](https://opensource.apple.com/source/Libc/Libc-262/i386/gen/strcat.c.auto.html) ) will always need to look for the last character not null before to append the text. 

Considering that you are already measuring the length for each one of the strings that we want to append for `user_plugins_dir`, why do not you store it inside `stack` ( in a `size_t` vector) and use `strcpy`)?

To be more clear, I am suggesting a simple reorganization of the code for : 

```c
strcpy(user_plugins_dir + vector_with_length[i], plugin_directories[i]);
```

Probably a preferred option would be to do math with a pointer advancing it after each copy. :thinking: 

The current algorithm works and it is fine, this is only a suggestion to avoid measure `PLUGINSD_MAX_DIRECTORIES` times the `user_plugins_dir` length.

Reply:
And I suggest to make it a function